core/src/main/scala/Jwt.scala

```diff
@@ -578,7 +578,7 @@ trait JwtCore[H, C] {
   protected def validate(header: H, claim: C, signature: String, options: JwtOptions) {
     if (options.signature && !signature.isEmpty) {
       throw new JwtNonEmptySignatureException()
-    } else if (options.signature && !extractAlgorithm(header).isEmpty) {
+    } else if (options.signature && extractAlgorithm(header).isDefined && !options.allowUnknownAlgos) {
```

Comment:
Let's try to rethink that part with those consideration:

- rely on the `signature` option rather than adding a new one.
- if a `JwtUnkwownAlgorithm` is extracted, throw a `JwtNonSupportedAlgorithm` otherwise a `JwtNonEmptyAlgorithmException `